server/src/main/java/org/opensearch/index/shard/RefreshListeners.java

```diff
@@ -90,23 +93,29 @@ public final class RefreshListeners implements ReferenceManager.RefreshListener,
      * We never set this to non-null while closed it {@code true}.
      */
     private volatile List<Tuple<Translog.Location, Consumer<Boolean>>> refreshListeners = null;
+
+    private volatile List<Tuple<Long, Consumer<Boolean>>> seqNoRefreshListeners = null;
```

Comment:
how are these seqNoRefreshListeners cleared when a replica is promoted? I think we'll need to auto fire them?